### PR TITLE
Add actions for certificate

### PR DESCRIPTION
**Pull request by gin0606**
I add actions for signing in CircleCI/TravisCI.
# Usage

``` ruby
# I set this ENVs in CircleCI Environment variables 
ENV['KEYCHAIN_NAME'] = "ios-build.keychain"
ENV['KEYCHAIN_PASSWORD'] = "very_important_password!"

create_keychain(
  default_keychain: true,
  unlock: true,
  timeout: 3600,
  lock_when_sleeps: true
)
import_certificate certificate_path: "certs/AppleWWDRCA.cer"
import_certificate certificate_path: "certs/dist.p12", certificate_password: ENV['MY_DIST_CERT_PASSWORD']
```
# Could you help me

sh method in `actions_helper.rb` is output args.
But that behavior is not good for sensitive data

this is `import_certificate` action log.

```
[21:57:43]: --------------------------------
[21:57:43]: --- Step: import_certificate ---
[21:57:43]: --------------------------------
[21:57:43]: [SHELL COMMAND]: security import certs/dist.p12 -k ~/Library/Keychains/ios-build.keychain -P very_important_password! -T /usr/bin/codesign
[21:57:43]: [SHELL]: 1 identity imported.
```

that output my `very_important_password!` in shell.
I don't want to output my passowrd.

Do you have something idea or best way?

### CODE REVIEW

**Comment by gin0606** [comment]
I will test for this actions :sunglasses:

File: lib/fastlane/actions/create_keychain.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          FastlaneCore::ConfigItem.new(key: :name,
+                                       env_name: "KEYCHAIN_NAME",
+                                       description: "Keychain name",
+                                       optional: false),
+          FastlaneCore::ConfigItem.new(key: :password,
+                                       env_name: "KEYCHAIN_PASSWORD",
+                                       description: "Password for the keychain",
+                                       optional: false),
+          FastlaneCore::ConfigItem.new(key: :default_keychain,
+                                       description: 'Lock keychain when the system sleeps',
```
Comment (KrauseFx) [inline lib/fastlane/actions/create_keychain.rb]: This seems to be the wrong description.

File: lib/fastlane/actions/create_keychain.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                                       is_string: false,
+                                       default_value: false),
+          FastlaneCore::ConfigItem.new(key: :unlock,
+                                       description: 'Lock keychain when the system sleeps',
```
Comment (KrauseFx) [inline lib/fastlane/actions/create_keychain.rb]: This seems to be the wrong description.

**Comment by KrauseFx** [comment]
@gin0606 thank you for submitting this pull request :+1: 

Could you show me an example usage on how you would use this new integration?

Thanks :smiley:

**Comment by gin0606** [comment]
@KrauseFx 
```ruby
desc "Submit a new Inhouse Build to Deploygate"
lane :inhouse do
  create_keychain(
    default_keychain: true,
    unlock: true,
    timeout: 3600,
    lock_when_sleeps: true
  )
  import_certificate certificate_path: "certs/AppleWWDRCA.cer"
  import_certificate certificate_path: "certs/dist.p12", certificate_password: ENV['CERT_PASSWORD']
  sigh(
    app_identifier: "my.app.identifier",
    team_id: "XXXXXXXXXX",
    adhoc: true,
  )
  ipa(
    scheme: "myapp",
    configuration: 'Release',
    embed: ENV["SIGH_PROFILE_FILE_NAME"],
  )
  deploygate
end
```

Thank you for checking my pull request :smile: 
This is my lane with new actions.
This actions to use an existing certificate.

`cert` action create iOS code signing certificates.
But I can't find a way to use an existing certificates.

File: lib/fastlane/actions/create_keychain.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
+module Fastlane
+  module Actions
+    class CreateKeychainAction < Action
+      def self.run(params)
+        commands = []
+        commands << sh("security create-keychain -p #{params[:password]} #{params[:name]}")
```
Comment (KrauseFx) [inline lib/fastlane/actions/create_keychain.rb]: What if `params[:password]` contains a space? I think we have to wrap it around `'`, right?

File: lib/fastlane/actions/create_keychain.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+        commands << sh("security default-keychain -s #{params[:name]}") if params[:default_keychain]
```
Comment (KrauseFx) [inline lib/fastlane/actions/create_keychain.rb]: Same here.

**Comment by KrauseFx** [comment]
This looks really good! The test coverage is amazing!

I added some comments: I think we have problems when the password or Keychain name contains `"` or spaces.

File: lib/fastlane/actions/create_keychain.rb (hunk 3 above)
Comment (gin0606) [inline lib/fastlane/actions/create_keychain.rb]: That is true! ill fix that :runner:

File: lib/fastlane/actions/create_keychain.rb (hunk 4 above)
Comment (gin0606) [inline lib/fastlane/actions/create_keychain.rb]: :ok_woman:

**Comment by gin0606** [comment]
@KrauseFx Thanks! :smiley: 

btw I think `actions_helper`s `sh` output is not good for this actions.
Please check https://github.com/KrauseFx/fastlane/pull/380#issue-91986349 *Could you help me* section :bow:

**Comment by KrauseFx** [comment]
Oh, you can just use `#{command}` instead of `sh command` and it won't be displayed in the log.

**Comment by gin0606** [comment]
Sorry, I don't know what you mean.

"`#{command}` instead of `sh command`" means ↓this? (probably not :confused: )
 Would you explain it again? 

```diff
-  sh "security create-keychain -p #{params[:password].shellescape} #{params[:name].shellescape}"
+  #{"security create-keychain -p #{params[:password].shellescape} #{params[:name].shellescape}"}
```

**Comment by KrauseFx** [comment]
Ohhh, sorry, I meant:

```ruby
`${command}`
```

**Comment by gin0606** [comment]
oops, i forgot that syntax :sweat_smile: 

I will use that! thanks a lot!

**Comment by gin0606** [comment]
I trid \`command\`. But that can't create keychain normality.
I trid to resolve that problem but i can't :cry: 

So I Add `Actions.sh` arg for output log. 
What do you think this change...? https://github.com/gin0606/fastlane/commit/12ae6ce9261a99ca585bb67d2a4c5de123c553a2

**Comment by KrauseFx** [comment]
Looks great, thanks again for working on this :+1: :smiley:

**Comment by gin0606** [comment]
Thank you!!! :laughing:

**Pull request merged**